Client/Frontend/Browser/Certificate Viewer/CertificateViewController.swift

```diff
@@ -0,0 +1,373 @@
+    }
+
+    if publicKeyInfo.keyUsage.contains(.VERIFY) {
+      keyUsages.append(Strings.CertificateViewer.verifyTitle)
+    }
+
+    if publicKeyInfo.keyUsage.contains(.WRAP) {
+      keyUsages.append(Strings.CertificateViewer.wrapTitle)
+    }
+
+    if publicKeyInfo.keyUsage.contains(.DERIVE) {
+      keyUsages.append(Strings.CertificateViewer.deriveTitle)
+    }
+
+    if publicKeyInfo.keyUsage.isEmpty || publicKeyInfo.keyUsage == .INVALID || publicKeyInfo.keyUsage.contains(.ANY) {
+      keyUsages.append(Strings.CertificateViewer.anyTitle)
+    }
+
+    let exponent = publicKeyInfo.type == .RSA && publicKeyInfo.exponent != 0 ? "\(publicKeyInfo.exponent)" : ""
+
+    // Ordered mapping
+    let mapping = [
+      KeyValue(key: Strings.CertificateViewer.algorithmTitle, value: algorithm),
+      KeyValue(key: Strings.CertificateViewer.parametersTitle, value: parameters),
+      KeyValue(key: Strings.CertificateViewer.publicKeyTitle, value: publicKey),
+      KeyValue(key: Strings.CertificateViewer.exponentTitle, value: exponent),
+      KeyValue(key: Strings.CertificateViewer.keySizeTitle, value: keySizeInBits),
+      KeyValue(key: Strings.CertificateViewer.keyUsageTitle, value: keyUsages.joined(separator: " ")),
+    ]
+
+    return mapping.compactMap({
+      $0.value.isEmpty
+        ? nil
+        : CertificateKeyValueView(
+          title: $0.key,
+          value: $0.value)
+    })
+  }
+
+  private func formattedSignature() -> String {
+    let signature = model.signature
+    return "\(signature.bytesSize) \(Strings.CertificateViewer.bytesUnitTitle) : \(BraveCertificateUtils.formatHex(signature.signatureHexEncoded))"
+  }
+
+  private func fingerprintViews() -> [CertificateKeyValueView] {
+    let sha256Fingerprint = model.sha256Fingerprint
+    let sha1Fingerprint = model.sha1Fingerprint
+
+    return [
+      CertificateKeyValueView(title: Strings.CertificateViewer.sha256Title, value: BraveCertificateUtils.formatHex(sha256Fingerprint.fingerprintHexEncoded)),
+      CertificateKeyValueView(title: Strings.CertificateViewer.sha1Title, value: BraveCertificateUtils.formatHex(sha1Fingerprint.fingerprintHexEncoded)),
+    ]
+  }
+
+  private struct KeyValue {
+    let key: String
+    let value: String
+  }
+}
+
+#if DEBUG
+struct CertificateView_Previews: PreviewProvider {
+  static var previews: some View {
+    let model = BraveCertificate(name: "leaf")!
+
+    CertificateView()
+      .environmentObject(model)
+  }
+}
+#endif
+
+class CertificateViewController: UIViewController, PopoverContentComponent {
+
+  init(certificate: BraveCertificateModel, evaluationError: String?) {
+    super.init(nibName: nil, bundle: nil)
+
+    let rootView = CertificateView(model: certificate,
+                                   evaluationError: evaluationError)
+    let controller = UIHostingController(rootView: rootView)
+
+    addChild(controller)
+    controller.view.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
This might be not needed if we use snapkit below, need to recheck if that's the case